GrafeasGroup/buttercup
# Add `find` command

Relevant issue: N/A

## Description:

Adds a `find` command to locate a submission. It takes a Reddit URL as parameter, this can be the link to the post on r/ToR, a partner subreddit or the link to a transcription. It will then try to find this submission and send it back to the user.

Because it takes multiple API calls to retrieve all the information, the bot posts the first data it gets immidiately and then updates the message periodically with the new data. This should make the bot more responsive.

This PR also includes a lot of utility to make it easier to work with the Blossom API. Some of these definitions can maybe be ported to `blossom_wrapper` later. Hopefully this work will make it easier to add other commands to the bot.

## Screenshots:

![The find executed on a completed post.](https://user-images.githubusercontent.com/13908946/121930165-573bbb00-cd42-11eb-9d60-63f6f5707678.png)

## Testing Instructions:

 Make sure you have the data for the Blossom wrapper properly defined, as shown in the wiki. You can then use the `/find` command with a Reddit URL.

## Checklist:
